phoenix-core/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -376,10 +377,14 @@ private static void preSplitSequenceTable(PhoenixConnection conn, int nSaltBucke
      * @return
      * @throws IOException
      */
-    private static org.apache.hadoop.hbase.client.Connection getHBaseConnection(Map<String, String> options)
+    private static org.apache.hadoop.hbase.client.Connection getHBaseConnection(Configuration config, Map<String, String> options)
             throws IOException {
 
         Configuration conf = HBaseConfiguration.create();
+
+        Configuration oldconf = HBaseConfiguration.create(config);
```

Comment:
Simply use 
`Configuration conf = HBaseConfiguration.create(config);`
and you'll automatically get all properties, including ZK.